engine/src/main/java/org/terasology/rendering/opengl/fbms/DisplayResolutionDependentFBOs.java

```diff
@@ -42,21 +42,19 @@
 
     private SwappableFBO gBufferPair;
 
-    private FBO.Dimensions fullScale = new FBO.Dimensions(-1, -1);
+    private FBO.Dimensions fullScale = new FBO.Dimensions();
```

Comment:
I seem to remember there are good reasons to initialize, as much as possible, within constructors. So, unless there is a specific reason to keep it here, I'd do it there.

Reply:
"I seem to remember there are good reasons to initialize, as much as possible, within constructors."
I couldn't find anything on this. Can you please elaborate a little?

And the reason why I'm doing this is simple: To strengthen the fact that we will have only and only one object of type Dimensions. As things happen, I've started looking into memory and GC problems in Terasology.. and returning new objects everywhere is a *huge* problem. As you might have gathered from my rants in Architecture, the way to solve it is simple: create one object, and keep setting the values again and again.

To reinforce this, I decided to shift this construction here, and retain the initialization (via `updateFullScale()`) in the constructor.

I would *really* like to keep it this way, although if you want, I can shift it to the constructor.

Reply:
I can't find solid evidence either anymore. There are a number of good posts and articles mentioning the advantages and disadvantages for the various forms of initialization but only personal opinions when comparing actually equivalent cases. I.e. it makes sense to have inline initialization if the alternative would be identical initializations in every constructor. It makes no big difference here given we have only one constructor and the compiler will move that initialization in the constructor anyway. 

Personally, I prefer to do it in the constructor (if I can) so that I have all initializations in one place, ideally in some kind of meaningful order.  But I can live with what you have now.